ERCS/erc-7570.md

```diff
@@ -0,0 +1,322 @@
+
+The `data` parameter MUST contain the Token Legal Status (“Token-Id”) represented by STATUS_XXX-constants mentioned above.
+
+
+```solidity
+function forceTransferFrom(address from, address to, uint256 amount, bytes calldata data) external;
+```
+
+#### freezeTokens
+Inherits from ERC-7551:
+
+This function MUST freeze `amount` tokens of `tokenHolder`. Frozen tokens cannot be transferred to other accounts. The function MUST emit a `TokensFrozen` event. The function MUST throw if `tokenHolder`'s active balance is less than `amount` (excluding already frozen tokens). The `data` parameter MAY be used to further document the action.
+
+The `data` parameter MUST contain the Token Legal Status (“Token-Id”) represented by STATUS_XXX-constants mentioned above.
+
+```solidity
+function freezeTokens(address tokenHolder, uint256 amount, bytes calldata data) external;
+```
+
+#### revokeRepresentative
+
+The `revokeRepresentative` function reverses the permissions previously granted to a representative by a token holder using the `authorizeRepresentative` function. It effectively removes the representative's authority to perform certain operations, such as `balanceReport` and `transfer`.
+
+The action of revoking a representative's authority ensures that the representative can no longer perform these tasks on behalf of the token holder. This functionality is critical in scenarios where the token holder wishes to regain full control over their operations or where a representative's duties are no longer required.
+
+
+#### unfreezeTokens
+Inherits from ERC-7551:
+
+This function MUST unfreeze `amount` tokens of `tokenHolder`. The function MUST emit a `TokensUnfrozen` event. The function MUST throw if `tokenHolder`'s frozen balance is less than `amount`. The `data` parameter MAY be used to further document the action.
+
+```solidity
+function unfreezeTokens(address tokenHolder, uint256 amount, bytes calldata data) external;
+```
+
+### Events
+
+#### AuthorizedOperator
+This event MUST be emitted when a representative has been authorized.
+
+```solidity
+    event AuthorizedRepresentative(
+        AuthorizedAction action,
+        address indexed signer,
+        address indexed representative,
+        address indexed tokenHolder
+    );
+```
+
+### RevokedOperator
+This event MUST be emitted when a representative has been revoked from authorization.
+
+```solidity
+    event RevokedRepresentative(
+        AuthorizedAction action,
+        address indexed signer,
+        address indexed representative,
+        address indexed tokenHolder
+    );
+```
+
+## Backwards Compatibility
+
+This new token standard does not impose any backward compatibility.
+
+## Copyright
+
+Copyright and related rights waived via CC0.
```

Comment:
Please use the exact waiver from [the template](https://github.com/ethereum/EIPs/blob/master/eip-template.md?plain=1#L120).